### PR TITLE
ICU-10879 Split out OBJECTS from Makefiles into separate files

### PR DESCRIPTION
**Pull request by foopoiuyt**
Work in progress for pulling OBJECTS definitions out from Makefile into separate files.

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-10879
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

**Comment by srl295** [comment]
@foopoiuyt can you sign the CLA? In any event , it looks interesting, but also note comment https://unicode-org.atlassian.net/browse/ICU-10879?focusedCommentId=58848

**Comment by foopoiuyt** [comment]
Thanks, I'll update soon (hopefully by tomorrow) for using globs in directories where the list is just all the .c/.cpp files.

**Comment by foopoiuyt** [comment]
I've updated most of the directories from list files to globs. There were a few cases (below) where there were multiple targets or where there were files I didn't see in the original Makefile list.  Right now, I've left these with lst files, but should I just leave those totally alone?

Hoping this'll pass the CI builds.

---

samples/layout: multiple builds with overlapping sources
samples/udata: two separate builds for reader and writer
test/cintltst: usrchdat.c says it's meant to be included not directly compiled
test/letest: multiple builds with overlapping sources
test/perf/leperf: many files not referenced from the Makefile.in
test/perf/normperf: two separate builds for normperf and simplenormperf
test/perf/ubrkperf: ubrkperfold.cpp only seemingly used from the dsp for separate target
tools/escapesrc: two separate build for escapesrc and tblgen
tools/genrb: two separate builds for genrb and derb
tools/icuinfo: plugin source seems like it may be optional
tools/tzcode: zdump.c not used from main build

**Comment by foopoiuyt** [comment]
> Firstly, (0thly ? :) thank you.
> 
> > I've updated most of the directories from list files to globs.
> 
> Sorry, why would you do that? This seems counterintuitive to the original point, which is to be able to build without parsing the Makefile. I think it's best NOT to make it a glob, but an explicit list.

Sorry, I think I may have misunderstood the context of the comment you pointed me to above or the meaning of the "Reasonable build systems should be able to use *.c and *.cpp." from that. I'd originally done it with the lists because I'd thought that was easier for the case of other build systems, but thought pointing me to that was a request to change that.

**Comment by foopoiuyt** [comment]
Removed the samples and tests for now and I believe the couple of tools mentioned leave it alone (and doing just genrb in its directory).

File: icu4c/source/tools/pkgdata/sources.lst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+pkgdata.cpp
```
[the remark this answers is not in the document]
Comment (srl295) [reply icu4c/source/tools/pkgdata/sources.lst:1]: ^ update, you caught this already.

File: icu4c/source/common/sources.lst [hunk 2]
```diff
@@ -0,0 +1,196 @@
+errorcode.cpp
```
Comment (markusicu) [inline icu4c/source/common/sources.lst:1]: please sort these .lst files

**Comment by foopoiuyt** [comment]
Renamed the source list files to .txt and ran unix sort on them.

**Comment by srl295** [comment]
Looking good, will merge tomorrow

**Pull request merged**